src/Input/InputParamEnumHandler.php

```
@@ -0,0 +1,91 @@
+        assert(is_a($type, BackedEnum::class, true));
+        if (! (is_int($props) || is_string($props))) {
+            if ($isDefaultAvailable) {
+                return $defaultValue;
+            }
+
+            throw new ParameterEnumTypeException($varName);
+        }
+
+        // Get the backing type of the enum
+        $backingType = $refEnum->getBackingType();
+        if ($backingType instanceof ReflectionNamedType && $backingType->getName() === 'int' && is_string($props)) {
```

PHPStan (2.1.15)

Instanceof between ReflectionNamedType and ReflectionNamedType will always evaluate to true.


(instanceof.alwaysTrue)